core/consensus/rhd/src/lib.rs

```diff
@@ -616,110 +632,118 @@ pub fn bft_threshold(n: usize) -> usize {
-	});
-
-	check_justification_signed_message(authorities, &message[..], just)
-}
-
-/// Check a prepare justification for a header hash.
-/// Provide all valid authorities.
-///
-/// On failure, returns the justification back.
-pub fn check_prepare_justification<B: Block>(authorities: &[AuthorityId], parent: B::Hash, just: UncheckedJustification<B::Hash>)
-	-> Result<PrepareJustification<B::Hash>, UncheckedJustification<B::Hash>>
-{
-	let message = Encode::encode(&PrimitiveMessage::<B, _> {
-		parent,
-		action: PrimitiveAction::Prepare(just.0.round_number as u32, just.0.digest.clone()),
-	});
-
-	check_justification_signed_message(authorities, &message[..], just)
-}
-
-/// Check proposal message signatures and authority.
-/// Provide all valid authorities.
-pub fn check_proposal<B: Block + Clone>(
-	authorities: &[AuthorityId],
-	parent_hash: &B::Hash,
-	propose: &::rhododendron::LocalizedProposal<B, B::Hash, AuthorityId, LocalizedSignature>)
-	-> Result<(), Error>
-{
-	if !authorities.contains(&propose.sender) {
-		return Err(ErrorKind::InvalidAuthority(propose.sender.into()).into());
-	}
-
-	let action_header = PrimitiveAction::ProposeHeader(propose.round_number as u32, propose.digest.clone());
-	let action_propose = PrimitiveAction::Propose(propose.round_number as u32, propose.proposal.clone());
-	check_action::<B>(action_header, parent_hash, &propose.digest_signature)?;
-	check_action::<B>(action_propose, parent_hash, &propose.full_signature)
-}
-
-/// Check vote message signatures and authority.
-/// Provide all valid authorities.
-pub fn check_vote<B: Block>(
-	authorities: &[AuthorityId],
-	parent_hash: &B::Hash,
-	vote: &::rhododendron::LocalizedVote<B::Hash, AuthorityId, LocalizedSignature>)
-	-> Result<(), Error>
-{
-	if !authorities.contains(&vote.sender) {
-		return Err(ErrorKind::InvalidAuthority(vote.sender.into()).into());
-	}
-
-	let action = match vote.vote {
-		::rhododendron::Vote::Prepare(r, ref h) => PrimitiveAction::Prepare(r as u32, h.clone()),
-		::rhododendron::Vote::Commit(r, ref h) => PrimitiveAction::Commit(r as u32, h.clone()),
-		::rhododendron::Vote::AdvanceRound(r) => PrimitiveAction::AdvanceRound(r as u32),
-	};
-	check_action::<B>(action, parent_hash, &vote.signature)
-}
-
-fn check_action<B: Block>(action: PrimitiveAction<B, B::Hash>, parent_hash: &B::Hash, sig: &LocalizedSignature) -> Result<(), Error> {
-	let primitive = PrimitiveMessage {
-		parent: parent_hash.clone(),
-		action,
-	};
-
-	let message = Encode::encode(&primitive);
-	if ed25519::verify_strong(&sig.signature, &message, &sig.signer) {
-		Ok(())
-	} else {
-		Err(ErrorKind::InvalidSignature(sig.signature.into(), sig.signer.clone().into()).into())
-	}
-}
-
-/// Sign a BFT message with the given key.
-pub fn sign_message<B: Block + Clone>(message: Message<B>, key: &ed25519::Pair, parent_hash: B::Hash) -> LocalizedMessage<B> {
+// fn check_justification_signed_message<H>(
```

Comment:
Why did you need this code entirely? If you need this code anytime in the future, you can just use git to get it back^^